### PR TITLE
Turn service branch into autosuggest on pre-need

### PR DESCRIPTION
**Pull request by jbalboni**
Adds fuzzy matching search and also fixes an issue with the listbox pushing down content below it.

screen shot 2017-12-18 at 11 44 29 am

### CODE REVIEW

File: src/js/common/schemaform/fields/AutosuggestField.jsx [hunk 1]
```diff
@@ -43,17 +55,43 @@ export default class AutosuggestWidget extends React.Component {
...
+      return suggestion.id;
+    }
+
+    return _.set('widget', 'autosuggest', suggestion);
+  }
+
+  getReviewLabel(formData, uiSchema, schema) {
+    const uiOptions = uiSchema['ui:options'];
+    if (!uiOptions.getOptions) {
+      return uiOptions.labels[formData] || schema.enumNames[schema.enum.indexOf(formData)];
```
[the remark this answers is not in the document]
Comment (cehsu) [reply src/js/common/schemaform/fields/AutosuggestField.jsx]: Also, sometimes when I repeated the same steps (typed "army" and hit enter), the form would navigate to the next page, and when I navigated back, there would be no information in the branch of service field.

**Review by cehsu — APPROVED** [review]
LGTM. Great work, the handling of typos provides a really smooth experience. Had a few minor questions/comments.
Also, it looks the browser aXe check indicated the following error in reference to the combobox (I'm not sure whether this is an aXe bug or an actual issue that we might want to resolve):
Ensures elements with an ARIA role that require child roles contain them

As well as the following recommendation:
To solve this violation, you need to:
Fix the following:
Required ARIA child role not present: listbox

Similarly, some of the keyboard navigation behavior is slightly different from what seems [standard](https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.0pattern/combobox-autocomplete-list.html) if we are following those guidelines for both the text box and listbox interaction, and :
-pressing escape should clear the textbox
-when traversing the form, if pressing escape or blurring away from the field while focus is on an option, the option is selected when it shouldn't be

**Review by rlbaker — APPROVED** [review]

File: src/sass/modules/_m-schemaform.scss [hunk 2]
```diff
@@ -508,6 +508,10 @@ legend.schemaform-label.schemaform-file-label {
   margin-top: 0;
 }
 
+.react-autosuggest__container {
```
[the remark this answers is not in the document]
Comment (jbalboni) [reply src/sass/modules/_m-schemaform.scss]: This is BEM style, which I like, but I don't think we should use it ourselves us the whole FE team wants to.

**Comment by jbalboni** [comment]
@cehsu This could use another look, I think I have the a11y issues addressed (and the other weirdness that was happening).

**Comment by cehsu** [comment]
Thanks @jbalboni , are the following ideas of interest as well (copied from previous comment)?

> Similarly, some of the keyboard navigation behavior is slightly different from what seems standard if we are following those guidelines for both the text box and listbox interaction, and :
> -pressing escape should clear the textbox
> -when traversing the form, if pressing escape or blurring away from the field while focus is on an option, the option is selected when it shouldn't be

**Review by cehsu — APPROVED** [review]
LGTM. Great job fixing those issues.

**Comment by jbalboni** [comment]
@cehsu Both of those issues should be fixed in what I pushed earlier, thanks for pointing them out!

**Pull request merged**